src/__tests__/labs/ViewLab.test.tsx

```diff
@@ -66,160 +66,149 @@ describe('View Lab', () => {
       },
     } as any)
 
-    let wrapper: any
-    await act(async () => {
-      wrapper = await mount(
-        <ButtonBarProvider.ButtonBarProvider>
-          <Provider store={store}>
-            <Router history={history}>
-              <Route path="/labs/:id">
-                <titleUtil.TitleProvider>
-                  <ViewLab />
-                </titleUtil.TitleProvider>
-              </Route>
-            </Router>
-          </Provider>
-        </ButtonBarProvider.ButtonBarProvider>,
-      )
-    })
-    wrapper.find(ViewLab).props().updateTitle = jest.fn()
-    wrapper.update()
-    return { wrapper: wrapper as ReactWrapper }
+    return render(
+      <ButtonBarProvider.ButtonBarProvider>
+        <Provider store={store}>
+          <Router history={history}>
+            <Route path="/labs/:id">
+              <titleUtil.TitleProvider>
+                <ViewLab />
+              </titleUtil.TitleProvider>
+            </Route>
+          </Router>
+        </Provider>
+      </ButtonBarProvider.ButtonBarProvider>,
+    )
   }
 
-  describe('title', () => {
-    it('should have called the useUpdateTitle hook', async () => {
-      const expectedLab = { ...mockLab } as Lab
-      await setup(expectedLab, [Permissions.ViewLab])
-      expect(titleUtil.useUpdateTitle).toHaveBeenCalled()
-    })
-  })
-
   describe('page content', () => {
     it('should display the patient full name for the for', async () => {
       const expectedLab = { ...mockLab } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-      const forPatientDiv = wrapper.find('.for-patient')
-
-      expect(forPatientDiv.find('h4').text().trim()).toEqual('labs.lab.for')
+      setup(expectedLab, [Permissions.ViewLab])
 
-      expect(forPatientDiv.find('h5').text().trim()).toEqual(mockPatient.fullName)
+      await waitFor(() => {
```

Comment:
Just curious, why waitFor here?

Reply:
You are waiting for data to be returned from the store so you have something to test against.

Reply:
I need to use `waitFor` or `findByText` to get some data instead of just a loader